pkg/internal/transform/spanner.go

```diff
@@ -106,6 +107,16 @@ func extractIP(b []uint8, size int) string {
 	return net.IP(b[:size]).String()
 }
 
+func extractTraceID(traceparent [55]byte) string {
+	// If traceparent was not set in eBPF, entire field should be all zeroes.
+	if traceparent[0] == 0 {
+		return ""
+	}
+	// See https://www.w3.org/TR/trace-context/#traceparent-header-field-values for format.
+	// Traceparent starts with version, e.g. "00-". TraceID is the following 32 hex digits
+	return string(traceparent[3:35])
```

Comment:
Can we please add validation here that the value actually is in the correct format? I'm worried about someone sending bad traceparent and we cause a panic and crash Beyla later.

Reply:
Yes thank you, this was just blindly grabbing it without any validation.  I've added some validation that ensures that at least the required amount of hex digits and dashes are there and in the right places.  I'm curious if a regex here would be much slower; it certainly would be fewer lines of code.
I'm not sure about the absolute indexing here, would it be worth it to add a set of constants to better define it?

Reply:
I think we only need a simple check when we extract the data, the hex validation will be done when we send the trace in that parse from hex call.

Reply:
In my view we only need to ensure we don't crash when we parse the traceId. It would be nice to do proper validation, but at the end of the day, if a customer is using bad traceIds from their product we can't do much.

Any extensive validation will cost us in CPU time, which we try to minimize, so with regards to that:
1. We shouldn't be doing extensive validation here, because this code is shared between traces and metrics. If we are producing metrics only, this validation will be unnecessary work. Perhaps it's best to move any validation and parsing into the trace code?
2. At the trace side perhaps we can look at the trace SDK to see if there's any code that does validate the traceparent? I would imagine if these rules were so strict that there would be a validator. Assuming this validator isn't super expensive to run.
3. Whatever we do, I'd trade not having proper validation with lowering the cost of instrumentation. In most cases we expect the traceIds to be correctly formatted, and when they are up to spec we will have correct ids. If the customer provided traceIds are violating 4), for example, what's the worst that can happen? Their traces may contain invalid traceIds, but they were invalid to begin with.

My initial concern wasn't with valid traceIds, it was with not causing a panic. If they send garbage in and we produce some invalid traceId numbers, I'm less concerned. Producing the traces is the slowest part of the product, If we add full on validation based on the rules above, it should be optional and configurable.

Reply:
In that case I think we can revert all of the validation that I've added, and just go back to lifting the bytes in the trace-ID position without further checks.  The traceparent [55]byte is either filled in or zeroed by eBPF, length is exactly 55 (according to instrumented Go map.)  The trace-id we find is already getting validated by the trace2.TraceIDFromHex function.  So I think it should be OK handling garbage values.
Thanks for the detailed explanation.

Reply:
OK, went back to just extracting it.  It does print out the traceID field that was found, but it doesn't use invalid ones and rejects with a specific enough debug message.